IdentityCore/src/workplacejoin/ios/MSIDWorkPlaceJoinUtil.m

```diff
@@ -132,4 +132,63 @@ + (SecIdentityRef)copyWPJIdentity:(__unused id<MSIDRequestContext>)context
     return identityRef;
 }
 
++ (nullable NSString*) getWPJStringData:(id<MSIDRequestContext>)context
+                          identifier:(nonnull NSString*)identifier
+                          error:(NSError*__nullable*__nullable)error
+{
+
+    NSString *teamId = [[MSIDKeychainUtil sharedInstance] teamId];
+
+    if (!teamId)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Encountered an error when reading teamID from keychain.");
+        return nil;
+    }
+    NSString *sharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin", teamId];
+
+    MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"Shared access group: %@.", sharedAccessGroup);
+
+    // Building dictionary to retrieve given identifier from the keychain
+    NSMutableDictionary *query = [[NSMutableDictionary alloc] init];
+    [query setObject:(__bridge id)(kSecClassGenericPassword) forKey:(__bridge id<NSCopying>)(kSecClass)];
+    [query setObject:identifier forKey:(__bridge id<NSCopying>)(kSecAttrAccount)];
+    [query setObject:(id)kCFBooleanTrue forKey:(__bridge id<NSCopying>)(kSecReturnAttributes)];
+    [query setObject:sharedAccessGroup forKey:(__bridge id)kSecAttrAccessGroup];
+
+    CFDictionaryRef result = nil;
+    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
+    if (status != errSecSuccess)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"String Data not found for identifier %@ in shared access group: %@ with error code:%d", identifier, sharedAccessGroup, (int) status);
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDKeychainErrorDomain, status, @"Failed to get items from keychain.", nil, nil, nil, context.correlationId, nil, NO);
+        }
+
+        if (result)
+        {
+            CFRelease(result);
+        }
+
+        return nil;
+    }
+
+    NSString *stringData = [(__bridge NSDictionary *)result objectForKey:(__bridge id)(kSecAttrService)];
+
+    if (result)
+    {
+        CFRelease(result);
+    }
+
+    if (!stringData || [[stringData stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]] length] == 0)
```

Comment:
I think we have shared utilities for this

Reply:
Thanks for pointing.